vignettes/preprocess.Rmd

```diff
@@ -0,0 +1,346 @@
+  var2 = c(0, 1, 1)
+)
+bin_df
+```
+
+In addition to the dataframe itself, you have to provide the name of the outcome column to `preprocess_data`. Here's what the preprocessed data looks like:
+
+```{r}
+# preprocess raw binary data
+preprocess_data(dataset = bin_df, outcome_colname = "outcome")
+```
+
+The output is a list of length two: `dat_transformed` which has the transformed data, and `grp_feats` which is a list of grouped features. Here, `grp_feats` is `NULL` because there are no perfectly correlated features (e.g. `c(0,1,0)` and `c(0,1,0)`, or `c(0,1,0)` and `c(1,0,1)` - see below for more details). 
+
+The first column (`var1`) in `dat_transformed` is changed to `var1_yes` that has zeros (no) and ones (yes). The values in the second column (`var2`) stay the same because it's already binary, but the name changes to `var2_1`. 
+
+## Categorical data
+
+On to non-binary categorical data:
+
+```{r}
+# raw categorical dataset
+cat_df <- data.frame(
+  outcome = c("normal", "normal", "cancer"),
+  var1 = c('a','b','c')
+  )
+cat_df
+```
+
+```{r}
+# preprocess raw categorical data
+preprocess_data(dataset = cat_df, outcome_colname = "outcome")
+```
+
+As you can see, this variable was split into 3 different columns - one for each type (a, b, and c). And again, `grp_feats` is `NULL`.
+
+## Continuous data
+
+Now, looking at continuous variables:
+
+```{r}
+# raw continuous dataset
+cont_df <- data.frame(
+  outcome = c("normal", "normal", "cancer"),
+  var1 = c(1,2,3)
+  )
+cont_df
+```
+
+```{r}
+# preprocess raw continuous data
+preprocess_data(dataset = cont_df, outcome_colname = "outcome")
+```
+
+Wow! Why did the numbers change? This is because the default is to normalize the data using `c("center","scale")`. While this is often best practice, you may not want to normalize the data, or you may want to normalize the data in a different way. If you don't want to normalize the data, you can use `method=NULL`:
+
+```{r}
+# preprocess raw continuous data, no normalization
+preprocess_data(dataset = cont_df, outcome_colname = "outcome", method = NULL)
+```
+
+You can also normalize the data in different ways. You can implement any method used by the [`caret::preProcess`](https://www.rdocumentation.org/packages/caret/versions/6.0-86/topics/preProcess) function (the `method` argument).
+
+Another feature of the `preprocess_data` function is that if you provide continuous variables as characters, they will be converted to numeric:
+
+```{r}
+# raw continuous dataset as characters
+cont_char_df <- data.frame(
+  outcome = c("normal", "normal", "cancer"),
+  var1 = c("1","2","3")
+  )
+cont_char_df
+```
+
+```{r}
+# preprocess raw continuous character data as numeric
+preprocess_data(dataset = cont_char_df, outcome_colname = "outcome")
+```
+
+If you don't want this to happen, and you want character data to remain character data even if it can be converted to numeric, you can use `to_numeric=FALSE` and they will be kept as categorical:
+
+```{r}
```

Comment:
Maybe say what happens at the end of this preprocessing?

Reply:
I added more of an explanation - I hope that helps. Let me know if there's something additional I should include! I also added a section at the very end about the next step being to run machine learning (because that's what I originally thought you meant, and I thought it was a good idea). We should add the link to the vignette on `run_ml` once we have it.